### PR TITLE
Fix Windows CI test failures by cleaning env vars

### PR DESCRIPTION
**Pull request by JohnStarich**
Adds a script to clean the test environment variables.
Uses a Go script for easy cross-OS portability.

~Currently under `./internal/cmd/cleanenv`. If it should be shared more widely, then `./cmd/cleanenv` may be more appropriate.~ Done.

Fixes https://github.com/agnivade/wasmbrowsertest/issues/40 and unblocks https://github.com/agnivade/wasmbrowsertest/pull/42

### CODE REVIEW

File: internal/cmd/cleanenv/main.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		exitCode := 1
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			exitCode = exitErr.ExitCode()
+		}
+		os.Exit(exitCode)
+	}
+}
+
+type App struct {
```
Comment (JohnStarich) [inline internal/cmd/cleanenv/main.go:32]: I designed this with extensibility and testability in mind, hence extra bits like this. It should help if we later decide to share the tool more widely.

File: README.md [hunk 2]
```diff
@@ -106,6 +106,25 @@ jobs:
       uses: actions/checkout@v2
 ```
 
+#### `total length of command line and environment variables exceeds limit`
```
Comment (agnivade) [inline README.md:109]: I'd like to have a separate section at the bottom `## Errors` and place this under that.

File: README.md [hunk 3]
```diff
@@ -106,6 +106,25 @@ jobs:
...
+
+If the error `total length of command line and environment variables exceeds limit` appears, then
+the current environment variables' total size has exceeded the maximum when executing Go Wasm binaries.
+
+To resolve this issue, install `cleanenv` and use it to prefix your command.
+
+For example, if this is the failing line in GitHub Actions:
```
Comment (agnivade) [inline README.md:116]: For example, if this is the command used:

File: README.md [hunk 4]
```diff
@@ -106,6 +106,25 @@ jobs:
...
+```bash
+go test -cover ./...
+```
+The new commands should be the following:
+```bash
+go install github.com/agnivade/wasmbrowsertest/cmd/cleanenv@latest
+cleanenv -remove-prefix GITHUB_ -- go test -cover ./...
+```
+
+The `cleanenv` command above removes all environment variables prefixed with `GITHUB_` before running the command after the `--`.
```
Comment (agnivade) [inline README.md:126]: Add a second sentence "You can use the flag multiple times to strip other environment variables as well."

File: cmd/cleanenv/main.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+			exitCode = exitErr.ExitCode()
+		}
+		os.Exit(exitCode)
+	}
+}
+
+type App struct {
+	Args        []string
+	Env         []string
+	Out, ErrOut io.Writer
```
Comment (agnivade) [inline cmd/cleanenv/main.go:35]: Let's name it `StdOut` and `ErrOut` for readability

File: cmd/cleanenv/main_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+			outputStr := strings.TrimSpace(output.String())
+			if outputStr != tc.expectOutput {
+				t.Errorf("Unexpected output: %q != %q", tc.expectOutput, outputStr)
+			}
+		})
+	}
+}
+
+func assertEqualError(t *testing.T, expected string, err error) {
```
Comment (agnivade) [inline cmd/cleanenv/main_test.go:96]: I have found https://github.com/carlmjohnson/be to be a nice and minimal error testing library. If you would prefer, we can use that or this is fine as well. Up to you.

File: cmd/cleanenv/main.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
+package main
```
Comment (agnivade) [inline cmd/cleanenv/main.go:1]: Maybe a package level comment as to why this tool is written might help for readers wondering what is this for.

**Review by agnivade — COMMENTED** [review]
In my testing, I can see that:

```
cleanenv -remove-prefix=JAVA -- GOOS=js GOARCH=wasm go test -exec=/home/agniva/go/bin/wasmbrowsertest -v
```

fails with `exec: "GOOS=js": executable file not found in $PATH`, and we need to move the env var declarations ahead like so:

```
GOOS=js GOARCH=wasm cleanenv -remove-prefix=JAVA -- go test -exec=/home/agniva/go/bin/wasmbrowsertest -v
```

Wondering if we can look into supporting this? Or if it's too much work, we can just go ahead and document it.

File: cmd/cleanenv/main_test.go (hunk 6 above)
Comment (JohnStarich) [reply cmd/cleanenv/main_test.go:96]: That's a neat one I hadn't seen before 👌 I've also used a fair bit of https://github.com/stretchr/testify to similar effect.

Up to you! I'm ok with avoiding the additional dependency for now. Seems like using an assertions library might be a good future refactor PR though 💯

File: cmd/cleanenv/main.go (hunk 7 above)
Comment (JohnStarich) [reply cmd/cleanenv/main.go:9]: Good idea. Added in 0563dc6

File: cmd/cleanenv/main.go (hunk 5 above)
Comment (JohnStarich) [reply cmd/cleanenv/main.go]: Updated in 1c76b55

File: README.md (hunk 4 above)
Comment (JohnStarich) [reply README.md:139]: Fixed in 6a93001

File: README.md (hunk 3 above)
Comment (JohnStarich) [reply README.md]: Fixed in 6a93001

File: README.md (hunk 2 above)
Comment (JohnStarich) [reply README.md]: Fixed in 1b4cd2fe40d0c605e868babbff6e66f4fef5105c

**Review by agnivade — APPROVED** [review]
Thanks!

**Pull request merged**